File: scripts/add_playlist_type_column.py
#!/usr/bin/env python3
"""
Add playlist_type column to playlists table if it doesn't exist.
This script should be run once to add the new column to the database.
"""

import os
import sys
from supabase import create_client, Client

def main():
    supabase_url = os.environ.get("SUPABASE_URL")
    supabase_key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")
    
    if not supabase_url or not supabase_key:
        print("Error: SUPABASE_URL and SUPABASE_SERVICE_ROLE_KEY must be set")
        sys.exit(1)
    
    supabase: Client = create_client(supabase_url, supabase_key)
    
    # Check if column exists by trying to query it
    try:
        result = supabase.table("playlists").select("playlist_type").limit(1).execute()
        print("Column 'playlist_type' already exists")
        return
    except Exception as e:
        if "playlist_type" in str(e).lower() or "column" in str(e).lower():
            print("Column doesn't exist, adding it...")
        else:
            print(f"Unexpected error checking column: {e}")
            sys.exit(1)
    
    # Add the column using raw SQL
    # Note: This requires the service role key to have permission to execute SQL
    try:
        # Use RPC if available, otherwise we'll need to use direct SQL
        # For Supabase, we typically need to use the SQL editor or migrations
        # Let's try using the REST API to execute SQL
        print("Note: This script requires the column to be added via Supabase SQL Editor.")
        print("Please run the following SQL in your Supabase SQL Editor:")
        print()
        print("ALTER TABLE playlists ADD COLUMN IF NOT EXISTS playlist_type TEXT;")
        print()
        print("After adding the column, run the ingestion script to populate the values:")
        print("python scripts/ingest_exports_to_supabase.py")
        sys.exit(0)
    except Exception as e:
        print(f"Error: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main()

File: scripts/add_playlist_type_column.ts
#!/usr/bin/env tsx
/**
 * Add playlist_type column to playlists table.
 * This script attempts to add the column using Supabase's REST API.
 * 
 * If this doesn't work, run the SQL migration file directly in Supabase SQL Editor:
 * migrations/add_playlist_type_column.sql
 */

import { createClient } from "@supabase/supabase-js";

function requireEnv(name: string): string {
  const v = process.env[name];
  if (!v) {
    console.error(`Error: ${name} environment variable is required`);
    process.exit(1);
  }
  return v;
}

async function main() {
  const url = requireEnv("NEXT_PUBLIC_SUPABASE_URL");
  const serviceKey = requireEnv("SUPABASE_SERVICE_ROLE_KEY");

  const supabase = createClient(url, serviceKey, {
    auth: { persistSession: false, autoRefreshToken: false },
  });

  // Check if column exists
  try {
    const { data, error } = await supabase
      .from("playlists")
      .select("playlist_type")
      .limit(1);

    if (!error) {
      console.log("✓ Column 'playlist_type' already exists");
      return;
    }
  } catch (e) {
    // Column doesn't exist, continue
  }

  console.log("Column 'playlist_type' does not exist.");
  console.log("\nTo add the column, please run the following SQL in your Supabase SQL Editor:");
  console.log("\n" + "=".repeat(60));
  console.log("ALTER TABLE playlists ADD COLUMN IF NOT EXISTS playlist_type TEXT;");
  console.log("=".repeat(60));
  console.log("\nAfter adding the column, run the ingestion script to populate values:");
  console.log("python scripts/ingest_exports_to_supabase.py");
}

main().catch((error) => {
  console.error("Error:", error);
  process.exit(1);
});
